postreise/plot/plot_bar_generation_vs_capacity.py

```diff
@@ -0,0 +1,303 @@
+        )
+
+    if not scenario_ids:
+        scenario_ids = []
+    if isinstance(scenario_ids, (int, str)):
+        scenario_ids = [scenario_ids]
+    if isinstance(scenario_names, str):
+        scenario_names = [scenario_names]
+    if scenario_names and len(scenario_names) != len(scenario_ids):
+        raise ValueError(
+            "ERROR: if scenario names are provided, number of scenario names must match number of scenario ids"
+        )
+    if not custom_data:
+        custom_data = {}
+    if len(scenario_ids) + len(custom_data) <= 1:
+        raise ValueError(
+            "ERROR: must include at least two scenario ids and/or custom data"
+        )
+    if isinstance(resource_types, str):
+        resource_types = [resource_types]
+    if not resource_labels:
+        resource_labels = dict()
+    if not isinstance(resource_labels, dict):
+        raise TypeError("ERROR: resource_labels should be a dictionary")
+
+    grid = Grid(["USA"])
+    id2loadzone = grid.id2zone
+    all_loadzone_data = {}
+    scenario_data = {}
+    for i, sid in enumerate(scenario_ids):
+        scenario = Scenario(sid)
+        all_loadzone_data[sid] = {
+            "gen": sum_generation_by_type_zone(scenario, time_range, time_zone).rename(
+                columns=id2loadzone
+            ),
+            "cap": sum_capacity_by_type_zone(scenario).rename(columns=id2loadzone),
+        }
+        scenario_data[sid] = {
+            "name": scenario_names[i] if scenario_names else scenario.info["name"],
+            "gen": {"label": "Generation", "unit": "TWh", "data": {}},
+            "cap": {"label": "Capacity", "unit": "GW", "data": {}},
+        }
+    for area, area_type in zip(areas, area_types):
+        loadzone_set = area_to_loadzone(grid, area, area_type)
+        for sid in scenario_ids:
+            scenario_data[sid]["gen"]["data"][area] = (
+                all_loadzone_data[sid]["gen"][loadzone_set]
+                .sum(axis=1)
+                .divide(1e6)
+                .astype("float")
+                .round(2)
+                .to_dict()
+            )
+            scenario_data[sid]["cap"]["data"][area] = (
+                all_loadzone_data[sid]["cap"][loadzone_set]
+                .sum(axis=1)
+                .divide(1e3)
+                .astype("float")
+                .round(2)
+                .to_dict()
+            )
+    for c_data in custom_data:
+        scenario_data[c_data["name"]] = c_data
+
+    for area in areas:
+        if not resource_types:
+            area_resource_types = sorted(
+                set(
+                    [
+                        r
+                        for sd in scenario_data.values()
+                        for side in ["gen", "cap"]
+                        for r, v in sd[side]["data"][area].items()
+                        if v > 0
+                    ]
+                )
+            )
```

Comment:
Why do you need the `set` here. You can sort on the list directly. Also, `resource_types` being a `list`, it will be nice to keep the same type in the `if` and `else`

Reply:
There might be duplicated values.

Reply:
Can we replace the `set`-wrapped list comprehension with a set comprehension instead to achieve the same thing? (and then sort outside of that)

Reply:
That should work. I will see.

Reply:
Done.